### PR TITLE
doc: Move MQTT sample to net folder

### PR DESCRIPTION
**Pull request by divipillai**
The MQTT sample is moved to networking folder
NCSDK-21055

### CODE REVIEW

File: doc/nrf/samples.rst [hunk 1]
```diff
@@ -27,6 +27,8 @@ Those samples are a good starting point for understanding how to put together yo
    samples/edge
    samples/gazell
    samples/matter
+   samples/multicore
```
Comment (jtguggedal) [inline doc/nrf/samples.rst:30]: Was this change intentional?

**Review by jtguggedal — APPROVED** [review]

**Review by umapraseeda — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]
